src/libraries/Pool.sol

```diff
@@ -603,4 +597,15 @@ library Pool {
             liquidityNet = info.liquidityNet;
         }
     }
+
+    /// @notice Add a signed liquidity delta to liquidity
+    /// @dev Must call toUint128 on the result to revert if it overflows or underflows. This is to allow optimizer inlining.
+    /// @param x The liquidity before change
+    /// @param y The delta by which liquidity should be changed
+    /// @return z The liquidity after the change, declared as uint256 to prevent implicit conversion when calling toUint128
+    function addDelta(uint128 x, int128 y) internal pure returns (uint256 z) {
```

Comment:
I see your point. Using `toUint128` within `addDelta` would increase gas on `addLiquidity` and `removeLiquidity` snapshots by 187. Because it disallows the optimizer to inline the function when it may revert.

> To save gas, it can inline short functions that do not contain control-flow branches or opcodes with side-effects.

Reference:
https://soliditylang.org/blog/2021/03/02/saving-gas-with-simple-inliner/

Reply:
I decided to inline `toUint128` within `addDelta` which ended up saving more gas. Thanks for the prompt. :grinning: